src/hotspot/share/opto/memnode.cpp

```diff
@@ -3455,6 +3457,25 @@ MemBarNode* MemBarNode::leading_membar() const {
   return mb;
 }
 
+void BlackholeNode::format(PhaseRegAlloc* ra, outputStream* st) const {
+  st->print("blackhole ");
+  bool first = true;
+  for (uint i = 0; i < req(); i++) {
+    Node* n = in(i);
+    if (n != NULL && OptoReg::is_valid(ra->get_reg_first(n))) {
+      if (first) {
+        first = false;
+      } else {
+        st->print(", ");
+      }
+      char buf[128];
```

Comment:
I'd introduce `PhaseRegAlloc::dump_register(Node*,outputStream*)` overload and put these 3 lines there. 

Then ADLC-generated code can be migrated to it later.

Reply:
That refactoring would make sense, but maybe as separate PR that does everything at once? So that we don't introduce changeset dependencies.

Reply:
I'm fine with it.